Content.Server/Stunnable/Systems/StunbatonSystem.cs

```diff
@@ -2,129 +2,105 @@
 using Content.Server.Power.EntitySystems;
 using Content.Server.Power.Events;
 using Content.Server.Stunnable.Components;
-using Content.Shared.Audio;
 using Content.Shared.Chemistry.EntitySystems;
 using Content.Shared.Damage.Events;
 using Content.Shared.Examine;
-using Content.Shared.Interaction.Events;
 using Content.Shared.Item;
 using Content.Shared.Popups;
 using Content.Shared.Stunnable;
-using Content.Shared.Toggleable;
-using Robust.Server.GameObjects;
 using Robust.Shared.Audio;
 using Robust.Shared.Audio.Systems;
-using Robust.Shared.Player;
 
 namespace Content.Server.Stunnable.Systems
 {
     public sealed class StunbatonSystem : SharedStunbatonSystem
     {
         [Dependency] private readonly SharedItemSystem _item = default!;
-        [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
         [Dependency] private readonly RiggableSystem _riggableSystem = default!;
         [Dependency] private readonly SharedPopupSystem _popup = default!;
         [Dependency] private readonly BatterySystem _battery = default!;
         [Dependency] private readonly SharedAudioSystem _audio = default!;
 
+
+        public const float PitchVariation = 0.25f;
+
         public override void Initialize()
         {
             base.Initialize();
 
-            SubscribeLocalEvent<StunbatonComponent, UseInHandEvent>(OnUseInHand);
             SubscribeLocalEvent<StunbatonComponent, ExaminedEvent>(OnExamined);
             SubscribeLocalEvent<StunbatonComponent, SolutionChangedEvent>(OnSolutionChange);
             SubscribeLocalEvent<StunbatonComponent, StaminaDamageOnHitAttemptEvent>(OnStaminaHitAttempt);
+            SubscribeLocalEvent<StunbatonComponent, ItemToggleActivateAttemptEvent>(TryTurnOn);
+            SubscribeLocalEvent<StunbatonComponent, ItemToggleDeactivatedEvent>(TurnOff);
         }
 
         private void OnStaminaHitAttempt(EntityUid uid, StunbatonComponent component, ref StaminaDamageOnHitAttemptEvent args)
         {
-            if (!component.Activated ||
-                !TryComp<BatteryComponent>(uid, out var battery) || !_battery.TryUseCharge(uid, component.EnergyPerUse, battery))
-            {
-                args.Cancelled = true;
-                return;
-            }
-
-            if (battery.CurrentCharge < component.EnergyPerUse)
-            {
-                _audio.PlayPvs(component.SparksSound, uid, AudioHelpers.WithVariation(0.25f));
-                TurnOff(uid, component);
-            }
-        }
-
-        private void OnUseInHand(EntityUid uid, StunbatonComponent comp, UseInHandEvent args)
-        {
-            if (comp.Activated)
-            {
-                TurnOff(uid, comp);
-            }
-            else
+            if (TryComp<ItemToggleComponent>(uid, out var itemToggleComp))
```

Comment:
have it return early + generally dont have comp in var names for obvious components
```cs
if (!TryComp<ItemToggleComponent>(uid, out var itemToggle))
    return;
```